### PR TITLE
add activity cancel docs

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

taking the opportunity to better document how activity cancellation works, per user request and confusion https://community.temporal.io/t/problems-cancelling-a-running-activity-from-parent-workflow/2169

## notes to reviewers

the writing we had before was hard to understand. I rewrote to emphasize clarity and connection of this to heartbeating.

see: 
- https://deploy-preview-563--mystifying-fermi-1bc096.netlify.app/docs/concepts/activities#activity-cancellation
- https://deploy-preview-563--mystifying-fermi-1bc096.netlify.app/docs/go/activities#cancellation

## Before

CleanShot 2021-08-06 at 00 26 20@2x

## After

CleanShot 2021-08-06 at 00 26 39@2x

### CODE REVIEW

**Review by tylerkohn — COMMENTED** [review]
@sw-yx looks good and thanks for calling out that cancellations are not immediate!

File: docs/concepts/activities.md [hunk 1]
```diff
@@ -50,11 +50,21 @@ Long running Activities can be used as a special case of leader election. Tempor
...
 
-A Workflow can request an Activity cancellation. Currently the only way for an Activity to learn that it was cancelled is through heart beating. The heartbeat request fails with a special error indicating that the Activity was cancelled. Then it is up to the Activity implementation to perform all the necessary cleanup and report that it is done with it. It is up to the Workflow implementation to decide if it wants to wait for the Activity cancellation confirmation or just proceed without waiting.
+import WhatIsActivityCancellation from '../content/what-is-activity-cancellation.md'
 
-Another common case for Activity heartbeat failure is that the Workflow that invoked it is in a completed state. In this case an Activity is expected to perform cleanup as well.
+<WhatIsActivityCancellation />
+
+:::note Cancellations are not immediate
+
+`ctx.Done()` is only signaled when a heartbeat is sent to the service.
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/concepts/activities.md:61]: yes but i didnt have time to figure out how better to phrase this

File: docs/concepts/activities.md [hunk 2]
```diff
@@ -50,11 +50,21 @@ Long running Activities can be used as a special case of leader election. Tempor
...
+Temporal's SDK throttles this so a heartbeat may not be sent to the service until 80% of the heartbeat timeout has elapsed.
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/concepts/activities.md:62]: temporal service, yes
```suggestion
Temporal's SDK throttles this so a heartbeat may not be sent to Temporal Server until 80% of the heartbeat timeout has elapsed.
```

File: docs/content/what-is-activity-cancellation.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+id: what-is-activity-cancellation
+title: What is Activity Cancellation?
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/content/what-is-activity-cancellation.md:3]: wew yeah this is a toughie. future TODO?

[the remark this answers is not in the document]
Comment (flossypurse) [reply docs/content/what-is-activity-cancellation.md:3]: Ya, we can TODO all of this.
Per our feature map, Cancellation is a stand alone feature. I think that makes a lot of sense for us - 
So, sounds like we need to dig into "what is Cancellation" and see what foundation that gives us.

File: docs/concepts/activities.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (flossypurse) [reply docs/concepts/activities.md:61]: OK - np. I guess this will flush out when we spend time in this area.

**Review by flossypurse — APPROVED** [review]

File: docs/content/what-is-activity-cancellation.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/content/what-is-activity-cancellation.md:3]: yes i like that

**Pull request merged**